gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,5 +1,17 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1-all.zip
```

Comment:
From https://github.com/opensearch-project/search-processor/issues/133 "There are other incompatibilities in the build for Gradle 8, so 7.6.1 is the only option to keep a working build."

Reply:
Given that the build passes on this PR, what are the other incompatibilities? The linked issue doesn't explain why we can't use 8.1, like OpenSearch core.

I will try removing the Jackson dependencies altogether though. If we can just pick them up transitively, that should save us time in the long run.

Reply:
Hmm.... nope. I was able to remove jackson-core and have it compile, but if I remove either of jackson-databind or jackson-annotations, the build fails.